internal/gitutil/git_test.go

```diff
@@ -0,0 +1,464 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitutil
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing/object"
+)
+
+const (
+	testRemoteRepo  = "./testdata/test-remote-repo"
+	testClonedRepo  = "./testdata/test-cloned-repo"
+	testUnknownRepo = "./testdata/unknown-repo"
+	testEmptyRepo   = "./testdata/test-folder"
+	testNewFile     = "example-git-file"
+)
+
+// setupGitRepo is a helper function used to setup the test git repo
+func setupRemoteRepos(t *testing.T) {
+	t.Helper()
+	// get the pwd
+	pwd := os.Getenv("PWD")
+	pathToLocalRepo := filepath.Join(pwd, testRemoteRepo)
+	pathToNewCloneRepo := filepath.Join(pwd, testClonedRepo)
+	// create the git repo
+	if err := createGitRepo(t, pathToLocalRepo); err != nil {
+		t.Fatalf("failed to create test repo: %v", err)
+	}
+	// clone the existing repo
+	if err := cloneGitRepo(t, pathToLocalRepo, pathToNewCloneRepo); err != nil {
+		t.Fatalf("failed to clone test repo: %v", err)
+	}
+}
+
+// createGitRepo is a helper function used to create a git repo for local testing
+func createGitRepo(t *testing.T, localRepo string) error {
+	t.Helper()
+	// create a git local repo
+	r, err := git.PlainInit(localRepo, false)
+	if err != nil {
+		t.Error("error while initializing git repo")
+		return err
+	}
+	// add a file to the repo
+	filename := filepath.Join(localRepo, testNewFile)
+	if err := os.WriteFile(filename, []byte("this is a test repo for gitutil package"), 0o644); err != nil {
+		t.Error("error while creating test file")
+		return err
+	}
+	// get the reference of the local repo via Worktree
+	w, err := r.Worktree()
+	if err != nil {
+		t.Error("error while getting worktree")
+		return err
+	}
+	// git add the file
+	if err := w.AddGlob("."); err != nil {
+		t.Error("error while adding file to git repo")
+		return err
+	}
+	// git commit the file
+	if _, err := w.Commit("init commit", &git.CommitOptions{
+		Author: &object.Signature{},
+	}); err != nil {
+		t.Error("error while committing file to git repo")
+		return err
+	}
+
+	return nil
+}
+
+// cloneGitRepo is a helper function used to clone the local repo to a new path
+func cloneGitRepo(t *testing.T, existingRemoteRepo, newCloneRepo string) error {
+	t.Helper()
+	// git clone using the local repo
+	r, err := git.PlainClone(newCloneRepo, false,
+		&git.CloneOptions{URL: existingRemoteRepo})
+	if err != nil {
+		return err
+	}
+	// add upstream remote
+	if _, err := r.CreateRemote(&config.RemoteConfig{
+		Name: "upstream",
+		URLs: []string{newCloneRepo},
+	}); err != nil {
+		return err
+	}
```

Comment:
ditto, you can t.Fatal here and don't have a return type

Reply:
Addressed with the suggested changes.